### ISSUE TITLE
YOLOV5-s model is a bit larger than pruned before with L1NormPruner

### ISSUE THREAD

**Issue by moonlightian** [microsoft/nni#5627]
**Describe the bug**:
After speedup yolov5-s with sparsity 0.2, and save state dict with torch.save(), I found that the model is a bit larger than pruned before.
![image](https://github.com/microsoft/nni/assets/49636631/e88e3406-1cdc-4929-a0c3-8a40599dd2cc)

**Environment**:
- NNI version: 3.0rc1
- Training service (local|remote|pai|aml|etc):
- Python version: 3.9
- PyTorch version: 1.12
- Cpu or cuda version: cuda11.3

**Reproduce the problem**
- Code|Example:
`import torch

from nni.common.concrete_trace_utils import concrete_trace
from nni.contrib.compression.pruning import L1NormPruner
from nni.contrib.compression.utils import auto_set_denpendency_group_ids
from nni.compression.pytorch.speedup.v2 import ModelSpeedup

model = torch.hub.load('ultralytics/yolov5', 'yolov5s', device='cpu')
#torch.save(model.state_dict(), './model/yolov5s.pt')
model(torch.rand([1, 3, 640, 640]))

# Exclude detection head?
config_list = [{
    'sparsity': 0.2,
    'op_types': ['Conv2d']
}, {
    'op_names':['model.model.model.24.m.0','model.model.model.24.m.1','model.model.model.24.m.2'],
    'exclude': True
}]
config_list = auto_set_denpendency_group_ids(model, config_list, torch.rand([1, 3, 640, 640]))

pruner = L1NormPruner(model, config_list)
masked_model, masks = pruner.compress()
pruner.unwrap_model()

graph_module = concrete_trace(model, (torch.rand([1, 3, 640, 640]), None, None, None))
ModelSpeedup(model, torch.rand([1, 3, 640, 640]), masks, graph_module=graph_module).speedup_model()
##torch.save(model.state_dict(), './model/yolov5s_pruned.pt')
model(torch.rand([1, 3, 640, 640]))

torch.save(model.state_dict(), '/tmp/modelhub/yolo/pruned_yolo.pth')`

- How to reproduce:
-python yolo_prune.py
